### PR TITLE
Add more unit-tests for stun_deserializer.c file

### PR DESCRIPTION
**Pull request by t2-g**
*Description of changes:*

This PR is for adding additional unit tests to improve coverage, by thoroughly testing various STUN API's .

*Test Steps:*

```
cmake -S test/unit-test -B build/ -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Debug -DBUILD_CLONE_SUBMODULES=ON -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -Werror -DNDEBUG -DLIBRARY_LOG_LEVEL=LOG_DEBUG'
make -C build all
cd build && ctest -V 
make coverage
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test/stun_deserializer/stun_deserializer_utest.c [hunk 1]
```diff
@@ -1649,3 +1649,429 @@ void test_StunDeserializer_ParseAttributeAddress_XorPeer_HappyPath( void )
...
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
+
+    result = StunDeserializer_FindAttribute( &( ctx ),
+                                             STUN_ATTRIBUTE_TYPE_ICE_CONTROLLED,
+                                             NULL );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_BAD_PARAM,
+                       result );
```
Comment (ActoryOu) [inline test/unit-test/stun_deserializer/stun_deserializer_utest.c]: I'd suggest to move this one to next test case `test_StunDeserializer_FindAttribute_BadParams` for better categorization.

File: test/unit-test/stun_deserializer/stun_deserializer_utest.c [hunk 2]
```diff
@@ -1649,3 +1649,429 @@ void test_StunDeserializer_ParseAttributeAddress_XorPeer_HappyPath( void )
...
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_NO_ATTRIBUTE_FOUND,
+                       result );
+
+    result = StunDeserializer_FindAttribute( &( ctx ),
+                                             STUN_ATTRIBUTE_TYPE_PRIORITY,
+                                             &( attribute ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
```
Comment (ActoryOu) [inline test/unit-test/stun_deserializer/stun_deserializer_utest.c:1782]: Please check the returned attribute with expected priority value.

File: test/unit-test/stun_deserializer/stun_deserializer_utest.c [hunk 3]
```diff
@@ -1649,3 +1649,429 @@ void test_StunDeserializer_ParseAttributeAddress_XorPeer_HappyPath( void )
...
+                                    &( header ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
+    /* First attribute is FINGERPRINT. */
+    result = StunDeserializer_GetNextAttribute( &( ctx ),
+                                                &( attribute ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
```
Comment (ActoryOu) [inline test/unit-test/stun_deserializer/stun_deserializer_utest.c:1896]: Please check the returned attribute value.

File: test/unit-test/stun_deserializer/stun_deserializer_utest.c [hunk 4]
```diff
@@ -1649,3 +1649,429 @@ void test_StunDeserializer_ParseAttributeAddress_XorPeer_HappyPath( void )
...
+                                    &( header ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
+    /* First attribute is Integrity. */
+    result = StunDeserializer_GetNextAttribute( &( ctx ),
+                                                &( attribute ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
```
Comment (ActoryOu) [inline test/unit-test/stun_deserializer/stun_deserializer_utest.c:1958]: Please check the returned attribute value.

File: test/unit-test/stun_deserializer/stun_deserializer_utest.c [hunk 5]
```diff
@@ -1649,3 +1649,429 @@ void test_StunDeserializer_ParseAttributeAddress_XorPeer_HappyPath( void )
...
+                                    &( header ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
+    /* First attribute is Message-Integrity. */
+    result = StunDeserializer_GetNextAttribute( &( ctx ),
+                                                &( attribute ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
```
Comment (ActoryOu) [inline test/unit-test/stun_deserializer/stun_deserializer_utest.c:2020]: Please check the returned attribute value.

File: test/unit-test/stun_deserializer/stun_deserializer_utest.c [hunk 6]
```diff
@@ -1649,3 +1649,429 @@ void test_StunDeserializer_ParseAttributeAddress_XorPeer_HappyPath( void )
...
+    /* Get the next attribute, which should be the Fingerprint attribute. */
+    result = StunDeserializer_GetNextAttribute( &( ctx ),
+                                                &( attribute ) );
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OK,
+                       result );
```
Comment (ActoryOu) [inline test/unit-test/stun_deserializer/stun_deserializer_utest.c:2029]: Please check the returned attribute value.

File: test/unit-test/stun_deserializer/stun_deserializer_utest.c [hunk 7]
```diff
@@ -1649,3 +1649,429 @@ void test_StunDeserializer_ParseAttributeAddress_XorPeer_HappyPath( void )
...
+
+    TEST_ASSERT_EQUAL( STUN_RESULT_OUT_OF_MEMORY,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+
+
+
```
Comment (ActoryOu) [inline test/unit-test/stun_deserializer/stun_deserializer_utest.c]:
```suggestion
```
Remove redundant empty lines (keep one and only one at the end).

Comment (t2-g) [reply test/unit-test/stun_deserializer/stun_deserializer_utest.c]: Sure, will keep that in mind

File: test/unit-test/stun_deserializer/stun_deserializer_utest.c (hunk 1 above)
Comment (t2-g) [reply test/unit-test/stun_deserializer/stun_deserializer_utest.c]: Yes, I have now added separate test cases.

**Review by ActoryOu — DISMISSED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**